### PR TITLE
EZP-28747: HTTP Cache purge when UserService\{CreateUser,CreateUserGroup,UpdateUser,UpdateUserGroup}Signal is emited

### PR DESCRIPTION
**Pull request by adamwojs**
> JIRA: https://jira.ez.no/browse/EZP-28747

## Description

This PR adds/fixes HTTP cache purging when the following signals are emitted:  

* `\eZ\Publish\Core\SignalSlot\Signal\UserService\CreateUserGroupSignal`
* `\eZ\Publish\Core\SignalSlot\Signal\UserService\CreateUserSignal`
* `\eZ\Publish\Core\SignalSlot\Signal\UserService\UpdateUserGroupSignal`
* `\eZ\Publish\Core\SignalSlot\Signal\UserService\UpdateUserSignal`

### CODE REVIEW

**Comment by adamwojs** [comment]
Review ping @andrerom @alongosz @bdunogier @natanael89

File: tests/SignalSlot/CreateUserGroupSlotTest.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public function createSignal()
+    {
+        return new CreateUserGroupSignal([
+            'userGroupId' => $this->contentId
+        ]);
+    }
+
+    public function getReceivedSignalClasses()
+    {
+        return ['eZ\Publish\Core\SignalSlot\Signal\UserService\CreateUserGroupSignal'];
```
Comment (natanael89) [inline tests/SignalSlot/CreateUserGroupSlotTest.php]: Could be `CreateUserGroupSignal::class` instead. Also true for some other places in this PR.

**Comment by andrerom** [comment]
Note: You can also rebase on top of 0.4 branch now.

File: tests/SignalSlot/CreateUserGroupSlotTest.php (hunk 1 above)
Comment (adamwojs) [reply tests/SignalSlot/CreateUserGroupSlotTest.php]: Done.

**Pull request merged**